### ISSUE TITLE
Hard coded font-face ressource in epub.css

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1316]
In epub.css there is a hardcoded font-face ressource url which throws an error in validation with epubcheck

```
#!css

@font-face {
    font-family: "LiberationNarrow";
    font-style: normal;
    font-weight: normal;
    src: url("res:///Data/fonts/LiberationNarrow-Regular.otf")
        format("opentype");
}
@font-face {
    font-family: "LiberationNarrow";
    font-style: oblique, italic;
    font-weight: normal;
    src: url("res:///Data/fonts/LiberationNarrow-Italic.otf")
        format("opentype");
}
@font-face {
    font-family: "LiberationNarrow";
    font-style: normal;
    font-weight: bold;
    src: url("res:///Data/fonts/LiberationNarrow-Bold.otf")
        format("opentype");
}
@font-face {
    font-family: "LiberationNarrow";
    font-style: oblique, italic;
    font-weight: bold;
    src: url("res:///Data/fonts/LiberationNarrow-BoldItalic.otf")
        format("opentype");
}
```

As ebook readers have default fonts which override this I think that this could be deleted entirely. At least the Liberation fonts declarations are not needed. They are referenced a few times in epub.css.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1316
- Originally reported by: Jan Ulrich Hasecke
- Originally created at: 2013-11-28T13:39:16.088

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-10 14:11:53+00:00_

So you would remove all the `@font-face` blocks?

**Comment by shimizukawa** [comment]
_From Jan Ulrich Hasecke on 2014-01-10 15:33:11+00:00_

I removed them without problems from my local Sphinx projects. I think they could be removed at all.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-10 19:12:19+00:00_

OK, thanks. Let's see if Roland has a comment.

**Comment by shimizukawa** [comment]
_From Roland Meister on 2014-01-10 20:49:01+00:00_

I am probably the only one that actually uses the settings.
So if there is a general consent to remove the direcitves, that's fine with me
(although that means work for me to reactivate the directives).
Also the settings only apply to a rather old type of reader, the newer
ones probably have other ways to select custom fonts.
But I do not want to lose the information about using @font-face,
so if the directives are removed, an example should be added
to the FAQ.

However, I'm not sure if the only reason is to get rid of a few
warnings from epubcheck or if there are other issues, e.g. with kindlegen.
In that case I know of at least one more problem with the css as it is now
(hidden elements). A custom kindle theme with it's own css file that
fixes the css issues and removes the @font-face directives seems to
be the more general solution. The css file can then also implement all
recommendations from amazon's kindle style guide.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-11 13:29:38+00:00_

Well, I guess that epubcheck passing (as much as possible) is not a bad goal. And since you admit you are probably the main audience for the font-face directives I vote for removing them.

**Comment by shimizukawa** [comment]
_From Jan Ulrich Hasecke on 2014-01-12 10:31:56+00:00_

As Amazons Kindle Format and ePub are de facto standards today I would vote for a solution that passes epubcheck and can be converted by kindlegen without problems. My problem lately was that the Kindle Direct Publishing Platform rejected a ePub file made by sphinx. Generally KDP accepts ePubs. I had to convert the ePub to Kindle via Calibre to upload the file. Kindlegen would be the other option to convert ePubs to mobi. 

I would guess that the font face directive only works when all fonts are included in the ePub. I am not sure whether Sphinx already copies fonts to the e-book directory.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-19 10:38:07+00:00_

Closes #1316: Remove hard-coded `font-face` resources from epub theme.

→ <<cset 97b3ac0187bd8941c7c9db94fd68a20fed7774f4>>
